### DISCUSSION TITLE
"Extract method" refactoring does not generate type hints for the new method arguments

### DISCUSSION THREAD

**Discussion by roman-g** [microsoft/pylance-release#2656] [Enhancement]
"Extract method" refactoring  is inconvenient, a developer needs to define types of all arguments manually, even if they had been known before the refactoring

As an example
```
def f(x: int, y: int) -> int:
    return x + y + 1;
```
Refactor... -> Extract method for `x + y` generates the following:

```
def f(x: int, y: int):
    return new_func(x, y) + 1;

def new_func(x, y):
    return x + y
```

Now I have to manually define type hints for all the arguments and the function return type.

It would be great if the result was

```
def new_func(x: int, y: int) -> int:
    return x + y
```

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by ouatu-ro** [comment]
Hello, this is an awesome idea! Recently opened the issue #5750 regarding exactly this! If there is a way for me to implement it, I would be happy to give it a try, although I am not certain what part of the code is accessible. At a quick glance, I wasn't able to find the underlying code behind `python.analysis.inlayHints.variableTypes` here on in the `pyright` repo, which seems most suited to handle this use-case too.

I would also be interested in alternative workflows for streamlined type hinting, although I doubt something would come close to this enhancement, any suggestion is appreciated.
